File: backend/src/main/java/com/ShopCart_FE_BE/dto/OrderDto.java
package com.ShopCart_FE_BE.dto;

import java.math.BigDecimal;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class OrderDto {
    private Long id;

    private String fullName;
    private String address;
    private String status;

    private String paymentMethod;
    private String paymentStatus;

    private BigDecimal totalAmount;
    private BigDecimal shippingFee;
    private Integer totalQuantity;

    private UserDto user;
    private List<OrderItemDto> items;
    private OrderCouponDto coupon;
}

File: frontend/src/libs/dto/OrderDto.ts
import type { OrderCouponDto } from "./OrderCouponDto";
import type { OrderItemDto } from "./OrderItemDto";
import type { UserDto } from "./UserDto";

export type OrderDto = {
    id: number;
    fullName: string;
    address: string;
    status: string;
    paymentMethod: string;
    paymentStatus: string;
    totalAmount: number;
    shippingFee: number;
    totalQuantity: number;
    user: UserDto;
    items: OrderItemDto[];
    coupon: OrderCouponDto;
}